packages/router-plugin/src/core/code-splitter/compilers.ts

```diff
@@ -449,6 +493,28 @@ export function compileCodeSplitVirtualRoute(opts: ParseAstOptions) {
 
   deadCodeElimination(ast)
 
+  // if there are exported identifiers, then we need to add a warning
+  // to the file to let the user know that the exported identifiers
+  // will not in the split file but in the original file, therefore
+  // increasing the bundle size
+  if (knownExportedIdents.size > 0) {
+    const list = Array.from(knownExportedIdents).reduce((str, ident) => {
+      str += `\n- ${ident}`
+      return str
+    }, '')
+
+    const warningMessage = `These exports from "${opts.filename.replace('?' + splitPrefix, '')}" are not being code-split and will increase your bundle size: ${list}\nThese should either have their export statements removed or be imported from another file that is not a route.`
```

Comment:
does this cause a problem with Start?
will Start properly work with some non-code-split routes?

it Start will not work, then we should throw a fatal error in case we are building for Start

Reply:
This is only capturing if the `component` or `loader` is being exported out of the route file. It'll detect it both for  Start and Router since this happens during dev and at build time (not at runtime).

This is the example that are checking for:
```ts
// this export ident here
export const loadUser = async () => {
	return { user: 'john' }
}

export const Route = createFileRoute('/login')({
	loader: loadUser
})
```

In this case, we can't code-split the `loadUser` fn, since other external files may be imported from this source file.

So it needs to remain as-is.

Reply:
understood, but is skipping a route from code splitting posing any problems in TanStack Start? i.e. could we just disable automatic code splitting for start and it would still work?

Reply:
> could we just disable automatic code splitting for start and it would still work?

We *could* choose to turn it off automatic code-splitting on our side, but currently, we force it on in the Start config. https://github.com/TanStack/router/blob/8fde2ef13d887f6657a0534c5f2e9e6216329121/packages/start/src/config/index.ts#L292